lib/spack/spack/test/cray_manifest.py

```diff
@@ -22,53 +22,7 @@
 import spack.spec
 import spack.store
 from spack.cray_manifest import compiler_from_entry, entries_to_specs
-
-example_x_json_str = """\
-{
-  "name": "packagex",
-  "hash": "hash-of-x",
-  "prefix": "/path/to/packagex-install/",
-  "version": "1.0",
-  "arch": {
-    "platform": "linux",
-    "platform_os": "centos8",
-    "target": {
-      "name": "haswell"
-    }
-  },
-  "compiler": {
-    "name": "gcc",
-    "version": "10.2.0.cray"
-  },
-  "dependencies": {
-    "packagey": {
-      "hash": "hash-of-y",
-      "type": ["link"]
-    }
-  },
-  "parameters": {
-    "precision": ["double", "float"]
-  }
-}
-"""
```

Comment:
@alalazo the changes to address your requests were minor, but I also wanted to update the unit tests, which took more effort:

* Module-level variables have been replaced with fixtures, specifically which draw from the `test_platform` fixture, which allows the unit tests to run with the new concretizer (my custom choice of os/target was confusing the new concretizer)
  * as a consequence of this it became convenient to move a `spack external find` test in here (since it was specifically testing whether Spack would collect packages from manifest files): `find_external_nonempty_default_manifest_dir`
* I added a new test `read_cray_manifest_add_compiler_failure` which confirms the behavior added in this PR

The diff in this file grew quite a bit, but that is largely from placing the manually-generated JSON at another level of indentation.